File: atcoder/abc190B_magic_3_.nim
# uninhm
# https://atcoder.jp/contests/abc190/tasks/abc190_b
# implementation

import strutils, sequtils

template for_else(a, b: string, code: untyped) =
  var flag {.inject.} = false
  code
  if flag:
    echo a
  else:
    echo b

var n, s, d: int64
(n, s, d) = stdin.readLine.split.map(parseInt)

for_else("Yes", "No"):
  var x, y: int64
  for i in 1 .. n:
    (x, y) = stdin.readLine.split.map(parseInt)
    if x < s and y > d:
      flag = true
      break

File: atcoder/abc190B_magic_3.hs
-- user: nel_tu_
-- path: https://atcoder.jp/contests/abc190/tasks/abc190_b
-- tags: implementation

import Data.Bool ( bool )

solve ([_, s, d] : spells) = any effective spells
    where effective [x, y] = x < s && y > d

input :: String -> [[Int]]
input = map (map read . words) . lines

main = interact $ bool "No" "Yes" . solve . input

